Tetrahedral_remeshing/include/CGAL/Tetrahedral_remeshing/internal/tetrahedral_remeshing_helpers.h

```diff
@@ -141,6 +141,189 @@ typename Tr::Geom_traits::FT min_dihedral_angle(const Tr& tr,
+  if (CGAL::NULL_VECTOR == ilij)
+    return Dihedral_angle_cosine(CGAL::POSITIVE, 1.,1.);
+
+  const FT num = scalar_product(ijik, ilij);
+  if(num == 0.)
+    return Dihedral_angle_cosine(CGAL::ZERO, 0.,1.);
+
+  const double sqden = CGAL::to_double(
+    scalar_product(ijik, ijik) * scalar_product(ilij, ilij));
+
+  return Dihedral_angle_cosine(CGAL::sign(num), CGAL::square(num), sqden);
+}
+
+template<typename Point, typename Geom_traits>
+Dihedral_angle_cosine max_cos_dihedral_angle(const Point& p,
+                                             const Point& q,
+                                             const Point& r,
+                                             const Point& s,
+                                             const Geom_traits& gt)
+{
+  Dihedral_angle_cosine a = cos_dihedral_angle(p, q, r, s, gt);
+  Dihedral_angle_cosine max_cos_dh = a;
+  if(max_cos_dh.is_one()) return max_cos_dh;
+
+  a = cos_dihedral_angle(p, r, s, q, gt);
+  if(max_cos_dh < a) max_cos_dh = a;
+  if (max_cos_dh.is_one()) return max_cos_dh;
+
+  a = cos_dihedral_angle(p, s, q, r, gt);
+  if (max_cos_dh < a) max_cos_dh = a;
+  if (max_cos_dh.is_one()) return max_cos_dh;
+
+  a = cos_dihedral_angle(q, r, p, s, gt);
+  if (max_cos_dh < a) max_cos_dh = a;
+  if (max_cos_dh.is_one()) return max_cos_dh;
+
+  a = cos_dihedral_angle(q, s, r, p, gt);
+  if (max_cos_dh < a) max_cos_dh = a;
+  if (max_cos_dh.is_one()) return max_cos_dh;
+
+  a = cos_dihedral_angle(r, s, p, q, gt);
+  if (max_cos_dh < a) max_cos_dh = a;
+
+  return max_cos_dh;
+}
+
+template<typename Tr>
+Dihedral_angle_cosine max_cos_dihedral_angle(const Tr& tr,
+                                             const typename Tr::Vertex_handle v0,
+                                             const typename Tr::Vertex_handle v1,
+                                             const typename Tr::Vertex_handle v2,
+                                             const typename Tr::Vertex_handle v3)
+{
+  return max_cos_dihedral_angle(point(v0->point()),
+                                point(v1->point()),
+                                point(v2->point()),
+                                point(v3->point()),
+                                tr.geom_traits());
+}
+
+template<typename Tr>
+Dihedral_angle_cosine max_cos_dihedral_angle(const Tr& tr,
+                                             const typename Tr::Cell_handle c)
+{
+  if (c->is_cache_valid())
+    return Dihedral_angle_cosine(CGAL::sign(c->sliver_value()),
+                                 CGAL::abs(c->sliver_value()), 1.);
+
+  Dihedral_angle_cosine cos_dh = max_cos_dihedral_angle(tr,
+                                                        c->vertex(0),
+                                                        c->vertex(1),
+                                                        c->vertex(2),
+                                                        c->vertex(3));
+  c->set_sliver_value(cos_dh.signed_square_value());
+  return cos_dh;
```

Comment:
warning : https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-5.2-Ic-12/Tetrahedral_remeshing/TestReport_lrineau_Ubuntu-latest-GCC6-CXX1z.gz

Reply:
@janetournois To remove that warning, remove the copy-constructor of `Dihedral_angle_cosine` and let the compiler generates the special member functions.